sound/soc/sof/pcm.c

```diff
@@ -394,6 +359,9 @@ static int sof_pcm_open(struct snd_pcm_substream *substream)
 	dev_dbg(sdev->dev, "pcm: open stream %d dir %d\n", spcm->pcm.pcm_id,
 		substream->stream);
 
+	/* Clear hw_params_upon_resume flag */
+	spcm->hw_params_upon_resume[substream->stream] = 0;
```

Comment:
the moral equivalent of .hw_params is .hw_free, wondering if we can have a case where the PCM is not closed but new parameters are passed?

Reply:
@plbossart I could reset it in hw_free. let me run some tests with it before I make the change.